### ISSUE TITLE
Bug: sam deploy - KeyError

### ISSUE THREAD

**Issue by NathanCFoster** [aws/aws-sam-cli#4813]
### Description:

When deploying using GitHub actions, a `Error: Status` pops is caught. We are deploying by adding [Python](https://github.com/actions/setup-python) and [SAM](https://github.com/marketplace/actions/setup-aws-sam-cli), then [configuring AWS credentials](https://github.com/marketplace/actions/configure-aws-credentials-action-for-github-actions)

### Steps to reproduce:

Create a basic GitHub actions workflow, with these actions:
```
    - name: Setup python
      uses: actions/setup-python@v4
      with:
        python-version: ${{ inputs.python_v }}
    - name: Setup sam
      uses: aws-actions/setup-sam@v2
      with:
        use-installer: true
    - name: Get Credentials
      uses: aws-actions/configure-aws-credentials@v1
      with:
        aws-region: ${{ inputs.region }}
        role-to-assume: ${{ inputs.role_to_assume }}
    - name: Deploy sam
      shell: bash
      working-directory: ${{ inputs.location }}
      run: |
        if [[ ${{ inputs.sam_config }} == "true" ]]
        then
        sam build && sam deploy
        else
        sam package ${{ inputs.sam_package_config }}
        sam deploy ${{ inputs.sam_config }}
        fi
```
then run the workflow and you should see an error.

### Observed result:

```
Waiting for changeset to be created..


Error: 'Status'
Traceback:
  File "click/core.py", line 1055, in main
  File "click/core.py", line 1657, in invoke
  File "click/core.py", line 1404, in invoke
  File "click/core.py", line 760, in invoke
  File "samcli/lib/cli_validation/image_repository_validation.py", line 92, in wrapped
  File "click/decorators.py", line 84, in new_func
  File "click/core.py", line 760, in invoke
  File "samcli/lib/telemetry/metric.py", line 183, in wrapped
  File "samcli/lib/telemetry/metric.py", line 148, in wrapped
  File "samcli/lib/utils/version_checker.py", line 42, in wrapped
  File "samcli/cli/main.py", line 92, in wrapper
  File "samcli/commands/_utils/cdk_support_decorators.py", line 39, in wrapped
  File "samcli/commands/_utils/command_exception_handler.py", line 93, in wrapper_command_exception_handler
  File "samcli/commands/_utils/command_exception_handler.py", line 73, in wrapper_command_exception_handler
  File "samcli/commands/deploy/command.py", line 223, in cli
  File "samcli/commands/deploy/command.py", line 361, in do_cli
  File "samcli/commands/deploy/deploy_context.py", line 181, in run
  File "samcli/commands/deploy/deploy_context.py", line 258, in deploy
  File "samcli/lib/deploy/deployer.py", line 528, in create_and_wait_for_changeset
  File "samcli/lib/deploy/deployer.py", line 317, in wait_for_changeset

An unexpected error was encountered while executing "sam deploy".
```
If I could debug this for you I would.

### Expected result:

`sam build && sam deploy` to not throw errors

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: Ubuntu 22.04.2 LTS
2. `sam --version`: 1.76.0
3. AWS region: us-east-1

```
{
  "version": "1.76.0",
  "system": {
    "python": "3.7.10",
    "os": "Linux-5.15.0-1033-azure-x86_64-with-debian-bookworm-sid"
  },
  "additional_dependencies": {
    "docker_engine": "20.10.22+azure-1",
    "aws_cdk": "Not available",
    "terraform": "1.3.9"
  }
}
```

Final message before error messages with `--debug` flag

```
Sending Telemetry: {'metrics': [{'commandRun': {'requestId': '2f19e869-8953-4580-a15d-3df02ae8a8b2', 'installationId': '950f2e41-f461-4ed0-a2e9-c6b1f7169f86', 'sessionId': 'f740b422-e3bd-48b3-b990-c9d468332f04', 'executionEnvironment': 'GitHubAction', 'ci': True, 'pyversion': '3.7.10', 'samcliVersion': '1.76.0', 'awsProfileProvided': False, 'debugFlagProvided': True, 'region': 'us-east-1', 'commandName': 'sam deploy', 'metricSpecificAttributes': {'projectType': 'CFN', 'gitOrigin': None, 'projectName': '8b13b436cfe354f9cfa77c7cb8e65b64d0078df86e243816027cc1b0ebf32783', 'initialCommit': None}, 'duration': 2588, 'exitReason': 'KeyError', 'exitCode': 255}}]}
```

**Comment by NathanCFoster** [comment]
After trying with the `with-installer` key set to false, it gave a little more info on the error:
```
Error: 'Status'
Traceback:
  File "/home/runner/work/_temp/setup-sam-Kv1CnL/.venv/lib/python3.9/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/runner/work/_temp/setup-sam-Kv1CnL/.venv/lib/python3.9/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/runner/work/_temp/setup-sam-Kv1CnL/.venv/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/runner/work/_temp/setup-sam-Kv1CnL/.venv/lib/python3.9/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/runner/work/_temp/setup-sam-Kv1CnL/.venv/lib/python3.9/site-packages/samcli/lib/cli_validation/image_repository_validation.py", line 92, in wrapped
    return func(*args, **kwargs)
  File "/home/runner/work/_temp/setup-sam-Kv1CnL/.venv/lib/python3.9/site-packages/click/decorators.py", line 84, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
  File "/home/runner/work/_temp/setup-sam-Kv1CnL/.venv/lib/python3.9/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/runner/work/_temp/setup-sam-Kv1CnL/.venv/lib/python3.9/site-packages/samcli/lib/telemetry/metric.py", line 183, in wrapped
    raise exception  # pylint: disable=raising-bad-type
  File "/home/runner/work/_temp/setup-sam-Kv1CnL/.venv/lib/python3.9/site-packages/samcli/lib/telemetry/metric.py", line 148, in wrapped
    return_value = func(*args, **kwargs)
  File "/home/runner/work/_temp/setup-sam-Kv1CnL/.venv/lib/python3.9/site-packages/samcli/lib/utils/version_checker.py", line 42, in wrapped
    actual_result = func(*args, **kwargs)
  File "/home/runner/work/_temp/setup-sam-Kv1CnL/.venv/lib/python3.9/site-packages/samcli/cli/main.py", line 92, in wrapper
    return func(*args, **kwargs)
  File "/home/runner/work/_temp/setup-sam-Kv1CnL/.venv/lib/python3.9/site-packages/samcli/commands/_utils/cdk_support_decorators.py", line 39, in wrapped
    return func(*args, **kwargs)
  File "/home/runner/work/_temp/setup-sam-Kv1CnL/.venv/lib/python3.9/site-packages/samcli/commands/_utils/command_exception_handler.py", line 93, in wrapper_command_exception_handler
    raise ex
  File "/home/runner/work/_temp/setup-sam-Kv1CnL/.venv/lib/python3.9/site-packages/samcli/commands/_utils/command_exception_handler.py", line 73, in wrapper_command_exception_handler
    return func(*args, **kwargs)
  File "/home/runner/work/_temp/setup-sam-Kv1CnL/.venv/lib/python3.9/site-packages/samcli/commands/deploy/command.py", line 194, in cli
    do_cli(
  File "/home/runner/work/_temp/setup-sam-Kv1CnL/.venv/lib/python3.9/site-packages/samcli/commands/deploy/command.py", line [361](https://github.com/ascending-llc/Operation/actions/runs/4325247351/jobs/7554317231#step:3:367), in do_cli
    deploy_context.run()
  File "/home/runner/work/_temp/setup-sam-Kv1CnL/.venv/lib/python3.9/site-packages/samcli/commands/deploy/deploy_context.py", line 167, in run
    return self.deploy(
  File "/home/runner/work/_temp/setup-sam-Kv1CnL/.venv/lib/python3.9/site-packages/samcli/commands/deploy/deploy_context.py", line 250, in deploy
    result, changeset_type = self.deployer.create_and_wait_for_changeset(
  File "/home/runner/work/_temp/setup-sam-Kv1CnL/.venv/lib/python3.9/site-packages/samcli/lib/deploy/deployer.py", line 528, in create_and_wait_for_changeset
    self.wait_for_changeset(result["Id"], stack_name)
  File "/home/runner/work/_temp/setup-sam-Kv1CnL/.venv/lib/python3.9/site-packages/samcli/lib/deploy/deployer.py", line 317, in wait_for_changeset
    status = resp["Status"]
```

**Comment by NathanCFoster** [comment]
Did a little digging and found this line at fault in the `aws-sam-cli/samcli/lib/deploy/deployer.py` file
`315 |        except botocore.exceptions.WaiterError as ex:`
not sure how an exception object doesn't have the status key, as the `last_response` key doesn't give a key error.

**Comment by NathanCFoster** [comment]
After doing some debugging it seems the error was an invalid perms error... `cloudformation:DescribeChangeSet` was required for the waiter to work however it seems like there isnt a proper error message for it.

**Comment by hnnasit** [comment]
Hi @NathanCFoster, thank you for opening the issue and providing the debug details. Seems like you have found the issue was due to a missing permission for your credentials and resolved it? I do agree there should be better error handling here.

**Comment by hawflau** [comment]
re-opening as it's not intended to be closed automatically

**Comment by hawflau** [comment]
Closing as the fix has been released in AWS SAM CLI v1.77.0
